iptables/iptables.go

```diff
@@ -330,12 +357,19 @@ func Exists(table Table, chain string, rule ...string) bool {
 	if supportsCOpt {
 		// if exit status is 0 then return true, the rule exists
 		_, err := Raw(append([]string{"-t", string(table), "-C", chain}, rule...)...)
+		if err != nil {
+			_, err = Raw(append([]string{"-t", string(table), "-C", chain}, append(rule, commentArgs...)...)...)
+		}
 		return err == nil
 	}
 
 	// parse "iptables -S" for the rule (it checks rules in a specific chain
 	// in a specific table and it is very unreliable)
-	return existsRaw(table, chain, rule...)
+	if existsRaw(table, chain, rule...) {
```

Comment:
```
root@1f98b1116996:/go/src/github.com/docker/libnetwork# iptables -t nat -A PREROUTING -p tcp -d 192.168.1.1 --dport 1234 -j DNAT --to-destination 172.17.0.1:4321 ! -i lo -m comment --comment "docker libnetwork"

root@1f98b1116996:/go/src/github.com/docker/libnetwork# iptables -t nat -S PREROUTING
-P PREROUTING ACCEPT
-A PREROUTING -d 192.168.1.1/32 ! -i lo -p tcp -m tcp --dport 1234 -m comment --comment "docker libnetwork" -j DNAT --to-destination 172.17.0.1:4321
```

It seems the comment position in `iptables -S` changed ? Also the double quote is missing in Exists check ?